src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+        RequestedDataSet: request.catalog,
+        radius: request.conesize,
+        use: 'catalog_overlay',
+        catalogProject: request.project
+    });
+    console.log('Does not dispatch yet ' + tReq);
+    //dispatchTableSearch(tReq);
+}
+
+/**
+ *
+ * @type {ClassicComponentClass<P>}
+ */
+class CatalogSelectView extends Component {
+
+    constructor(props) {
+        super(props);
+        if (!isEmpty(catmaster)) {
+            this.state = {...this.state, master: {catmaster, cols}};
+        } else {
+            this.state = {...this.state, master: {}};
+        }
+    }
+
+    componentWillMount() {
+        if (isEmpty(catmaster)) {
+            this.loadMasterCatalogTable();
+        }
+    }
+
+    /**
+     * Fetch master catalog table with search id = 'irsaCatalogMasterTable'
+     * 'tableModel.tableData.data' returned in the promise is an array of elements which value is refered to the ith column name
+     * from tableModel.tableData.columns
+     * result is set in the state as 'master' object {catmaster, cols}
+     * @see setMaster
+     * @returns {Void} fetch master table and once fetched, set state with it
+     */
+    loadMasterCatalogTable() {
+
+        const request = {id: 'irsaCatalogMasterTable'}; //Fetch master table
+        TblUtil.doFetchTable(request).then((tableModel) => {
+
+            var data = tableModel.tableData.data;
+
+            cols = tableModel.tableData.columns;
+
+            var projects = [], subprojects = {}, option = [];
+            var i;
+            for (i = 0; i < data.length; i++) {
+                if (projects.indexOf(data[i][0]) < 0) {
+                    projects.push(data[i][0]);
+                }
+            }
+            for (i = 0; i < projects.length; i++) {
+                option = [];
+                const tmp = [];
+                data.forEach((e) => {
+                    if (e[0] === projects[i]) {
+                        if (option.indexOf(e[1]) < 0) {
+                            option.push(e[1]);
+                            var obj = {
+                                label: e[1],
+                                value: e[1],
+                                proj: projects[i]
+                            };
+                            tmp.push(obj);
+                        }
+                    }
+                });
+                subprojects[projects[i]] = tmp;
+
+
+            }
+            var o = 0;
+            for (i = 0; i < projects.length; i++) {
+                var opt = subprojects[projects[i]];
+                const tmp = [];
+                for (o = 0; o < opt.length; o++) {
+                    var myobj = {};
+                    myobj.project = projects[i];
+                    myobj.subproject = opt[o];
+                    var tab = data.filter((e) => {
+                        if (e[0] === projects[i] && e[1] === opt[o].value) {
+                            return e;
+                        }
+                    });
```

Comment:
This might need to be a `filter` not a `map`, i think it might be filling tab with some `undefined` elements.